PyDSS/cli/reports.py

```diff
@@ -0,0 +1,141 @@
+"""
+CLI to run a PyDSS project
+"""
+
+import ast
+import logging
+import os
+import sys
+import json
+import click
+
+from PyDSS.pydss_project import PyDssProject
+from PyDSS.loggers import setup_logging
+from PyDSS.utils.utils import get_cli_string, make_human_readable_size
+from PyDSS.common import SIMULATION_SETTINGS_FILENAME
+from terminaltables import SingleTable
+
+logger = logging.getLogger(__name__)
+
+@click.argument(
+    "project-path",
+)
+
+@click.option(
+    "-l", "--list",
+    help="List all reports for a given project path",
+    is_flag=True,
+    default=False,
+    show_default=True,
+)
+
+@click.option(
+    "-p", "--project",
+    required=False,
+    help="PyDSS project name.",
+)
+
+@click.option(
+    "-i", "--index",
+    help="View report by index (use -l flag to see list of available reports)",
+    default=0,
+    show_default=True,
+)
+
+@click.option(
+    "-s", "--scenario",
+    required=False,
+    help="PyDSS scenario name.",
+)
+
+@click.option(
+    "-r", "--report",
+    required=False,
+    help="PyDSS report name.",
+)
+@click.command()
+
+def reports(project_path, list=False, project=None, scenario=None, report=None, index=0):
+    """Explore and print PyDSS reports."""
+    assert not (list and index), "Both 'list' and 'index' options cannot be set to true at the same time"
+    assert os.path.exists(project_path), "The provided project path {} does not exist".format(project_path)
+    logsPath = os.path.join(project_path, "Logs")
+    assert os.path.exists(logsPath), "No Logs folder in the provided project path.".format(project_path)
+    reportList = getAvailableReports(logsPath)
+    if list:
+        Table = SingleTable(reportList, title="Available PyDSS reports")
+        print("")
+        print(Table.table)
+    elif index:
+        idx, projectName, ScenarioName, ReportName = reportList[index]
+        printReport(logsPath, projectName, ScenarioName, ReportName)
+    elif project:
+        for dx, projectName, ScenarioName, ReportName in reportList[1:]:
+            if projectName == project:
+                if scenario is None or scenario == ScenarioName:
+                    if report is None or report == ReportName:
+                        printReport(logsPath, projectName, ScenarioName, ReportName)
+
+def printReport(logsPath, project, scenario, report):
+    fileName = "{}__{}__reports.log".format(project, scenario)
+    filePath = os.path.join(logsPath, fileName)
+    assert os.path.exists(filePath), "Report {} for project: {} / scenario: {} does not exist".format(
+        report, project, scenario
+    )
+    with open(os.path.join(logsPath, fileName), "r") as f:
+        for line in f:
```

Comment:
Something went wrong here.  This won't run.  In my comment in this area I was suggesting that you use this instead of `f.readlines()`.